cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/DefaultProvisionerContext.java

```diff
@@ -125,4 +132,18 @@ public ProvisionerMetrics getMetrics(Map<String, String> context) {
     tags.put(Constants.Metrics.Tag.PROVISIONER, provisionerName);
     return new DefaultProvisionerMetrics(metricsCollectionService.getContext(tags));
   }
+
+  @Override
+  public <T> CompletionStage<T> execute(Callable<T> callable) {
+    CompletableFuture<T> result = new CompletableFuture<>();
+
+    executor.execute(() -> {
+      try {
+        result.complete(callable.call());
+      } catch (Exception e) {
```

Comment:
Do we want to catch Throwable? What is our strategy on handling something like StackOverflowError here?

Reply:
We can also use standard wrapping approach with something like this:
```
CompletableFuture.supplyAsync(executor,  () -> {
 try {
  result.complete(callable.call());
 } catch (Exception e) {
  throw new CompletionException(e);
 }
});
```
It would produce same result, but we would rely on JDK to ensure any exit from function will complete the future without explicit Throwable catching.

Reply:
Change to catch Throwable is simpler. The supplyAsync doesn't really buy us anything in this case.

Reply:
Generally speak, catching Throwable is not desirable. But in this case is necessary to complete the future returned.

Reply:
Yep, that's the main reason I proposed supplyAsync. I don't like to catch Throwable either.